### PR TITLE
Added GivenName and FamilyName claim types that LinkedIn returns.

### PR DESCRIPTION
**Pull request by ericgreenmix**
Also moved formattedName to the Name claim type and removed the UserName property because LinkedIn doesn't return a username.

### CODE REVIEW

File: Owin.Security.Providers/LinkedIn/LinkedInAuthenticationHandler.cs [hunk 1]
```diff
@@ -104,17 +104,21 @@ public LinkedInAuthenticationHandler(HttpClient httpClient, ILogger logger)
...
+                {
+                    context.Identity.AddClaim(new Claim(ClaimTypes.GivenName, context.GivenName, XmlSchemaString, Options.AuthenticationType));
+                }
+                if (!string.IsNullOrEmpty(context.FamilyName))
+                {
+                    context.Identity.AddClaim(new Claim(ClaimTypes.Surname, context.FamilyName, XmlSchemaString, Options.AuthenticationType));
+                }
                 if (!string.IsNullOrEmpty(context.Name))
                 {
-                    context.Identity.AddClaim(new Claim("urn:linkedin:name", context.Name, XmlSchemaString, Options.AuthenticationType));
```
Comment (jerriep) [inline Owin.Security.Providers/LinkedIn/LinkedInAuthenticationHandler.cs:117]: Please do not remove existing claims. You may break backwards compatibility for some people. Feel free to add a `ClaimTypes.Name` claim over and above the existing one.

File: Owin.Security.Providers/LinkedIn/LinkedInAuthenticationHandler.cs [hunk 2]
```diff
@@ -104,17 +104,21 @@ public LinkedInAuthenticationHandler(HttpClient httpClient, ILogger logger)
                 {
                     context.Identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, context.Id, XmlSchemaString, Options.AuthenticationType));
                 }
-                if (!string.IsNullOrEmpty(context.UserName))
```
Comment (jerriep) [inline Owin.Security.Providers/LinkedIn/LinkedInAuthenticationHandler.cs:107]: Please do not remove existing claims.

**Comment by jerriep** [comment]
@ericgreenmix Any feedback on those issues?

**Comment by ericgreenmix** [comment]
@jerriep I disagree about breaking backwards compatibility in this case. The current code was building a quasi "username" by removing the space in the LinkedIn `formattedName` claim type and then on top of that sticking it in the claim type for a name. Then creating a `urn:linkedin:name` to stick the `formattedName` from LI. This is extremely misleading when implementing the provider. Any work that modifies the base of what LI returns, in this case building a "username", should be done outside the context of the provider.

**Comment by jerriep** [comment]
@ericgreenmix If someone used the `urn:linkedin:name` claim before, then it will not be available after your changes. That is breaking backwards compatibility. No doubt about it. 

Please put it back or otherwise I cannot accept your PR.

**Comment by ericgreenmix** [comment]
@jerriep I added the `urn:linkedin:name` back.

File: Owin.Security.Providers/LinkedIn/Provider/LinkedInAuthenticatedContext.cs [hunk 3]
```diff
@@ -69,12 +70,17 @@ public LinkedInAuthenticatedContext(IOwinContext context, JObject user, string a
...
+        /// <summary>
+        /// Get the user's first name
+        /// </summary>
+        public string GivenName { get; private set; }
 
         /// <summary>
-        /// Gets the LinkedIn username
+        /// Get the user's last name
         /// </summary>
-        public string UserName { get; private set; }
```
Comment (jerriep) [inline Owin.Security.Providers/LinkedIn/Provider/LinkedInAuthenticatedContext.cs:77]: Please do not remove things which can break other people's code. Put this property back

Comment (ericgreenmix) [inline Owin.Security.Providers/LinkedIn/Provider/LinkedInAuthenticatedContext.cs:77]: The UserName property is not being used anymore. It was being set as the `name` claim type. The name claim is actually set to the LI `formatedName` now, which is the correct approach to take. If someone was using the name claim to get this arbitrary username, which was just the LI `formattedName` without spaces, then they will need to strip spaces from the name claim on their own now (the way if should have been done in the first place). That will be the only "backwards compatibility" that gets broken. 

It was incredibly confusing to me when using this package as to why the Name claim was incorrect from what LI was returning. This change allows future implementations of the library to be accurate with what comes from LI, so users of the package don't have to go digging through source code like I did.

Comment (jerriep) [inline Owin.Security.Providers/LinkedIn/Provider/LinkedInAuthenticatedContext.cs:77]: The library may not be using the `UserName` property internally anymore, but you have no guarantee that no one else is using it. 

We can go around in circles for days like this, but I am not going to accept a PR which has the potential of breaking other people's code. In the end I am the one who will get shout at when I release a minor version which does not have backwards compatibility, not you. 

So please put it back. Feel free however to mark it as `[Obsolete]` with a message as to the correct property to use.

**Comment by ericgreenmix** [comment]
I went ahead and added the username back and marked it as obsolete, as per your suggestion. That seems like an appropriate fix.

File: Owin.Security.Providers/LinkedIn/Provider/LinkedInAuthenticatedContext.cs [hunk 4]
```diff
@@ -36,8 +36,9 @@ public LinkedInAuthenticatedContext(IOwinContext context, JObject user, string a
 
             Id = TryGetValue(user, "id");
             Name = TryGetValue(user, "formattedName");
+            FamilyName = TryGetValue(user, "lastName");
+            GivenName = TryGetValue(user, "firstName");
             Link = TryGetValue(user, "publicProfileUrl");
-            UserName = TryGetValue(user, "formattedName").Replace(" ", "");
```
Comment (jerriep) [inline Owin.Security.Providers/LinkedIn/Provider/LinkedInAuthenticatedContext.cs:40]: Please restore this as well. It makes no sense putting back the `UserName` property if its value is always going to be null :) Then it still breaks people's code.

**Comment by jerriep** [comment]
Hi @ericgreenmix , sorry I missed the last commit you did which is why I have not merged this before.

Thanks for your patience in sorting this out :)

**Pull request merged**